src/pages/Main/MainCarousel.js

```diff
@@ -0,0 +1,70 @@
+import React, { Component } from 'react';
+import ImgCom from './ImgCom';
+
+class MainCarousel extends Component {
+  constructor() {
+    super();
+    this.state = {
+      firstImg: 0,
+      sliderArr: [],
+      info: {},
+    };
+  }
+
+  componentDidMount() {
+    fetch('http://localhost:3000/data/visualData.json')
+      .then(res => res.json())
+      .then(data => {
+        this.setState({
+          sliderArr: data,
+        });
+      });
+  }
+
+  slideLeft = () => {
+    const { firstImg } = this.state;
+    firstImg === 0
+      ? this.setState({ firstImg: 0 })
+      : this.setState({ firstImg: firstImg + 100 });
+  };
+
+  slideRight = () => {
+    const { firstImg, sliderArr } = this.state;
+    firstImg === -100 * (sliderArr.length - 1)
+      ? this.setState({ firstImg: 0 })
+      : this.setState({ firstImg: firstImg - 100 });
+  };
+
+  render() {
+    const { firstImg, sliderArr } = this.state;
+
+    return (
+      <>
+        <div className="visualWrap">
+          <div className="visualSlide">
+            <ul
+              className="imgWrap"
+              style={{ transform: `translateX(${firstImg}%)` }}
+            >
+              {sliderArr &&
+                sliderArr.map(ele => {
+                  return (
+                    <li className="slide">
+                      <ImgCom key={ele.id} alt={ele.alt} src={ele.src} />
+                    </li>
+                  );
```

Comment:
- ele 보다는 sliderElement 와 같이 명확하게 네이밍 하는 습관을 들여주세요!
- key 는 map 이 리턴하는 jsx 에서 가장 바깥 태그에 부여해 주셔야 합니다.
```suggestion
                sliderArr.map(sliderElement => {
                  return (
                    <li className="slide" key={sliderElement.id}>
                      <ImgCom alt={sliderElement.alt} src={sliderElement.src} />
                    </li>
                  );
```